mesonbuild/modules/selinux.py

```diff
@@ -0,0 +1,149 @@
+        KwargInfo('install_dir', (str, NoneType)),
+        KwargInfo('name', (str, NoneType)),
+        KwargInfo('mls', bool, default=False),
+        KwargInfo('type', (str, NoneType)),
+        KwargInfo('distro', (str, NoneType)),
+        KwargInfo('direct_initrc', (bool, NoneType)),
+    )
+    def package(self, state: ModuleState, args: T.Tuple[str], kwargs: TYPE_kwargs) -> ModuleReturnValue:
+        sharedir = os.path.join(state.environment.get_prefix(), state.environment.get_datadir(), 'selinux')
+        tools = self._find_tools(state)
+        for tool_name, tool in tools.items():
+            if not tool or not tool.found():
+                raise MesonException(f'SELinux tool {tool_name} not found')
+
+        package_name = args[0]
+        ppfile = args[0] + '.pp'
+        depfile = args[0] + '.d'
+
+        te_file = T.cast(T.Optional[T.Union[str, mesonlib.File]], kwargs.get('te_file'))
+        te_file = self._resolve_file_path(state, te_file)
+        if_file = T.cast(T.Optional[T.Union[str, mesonlib.File]], kwargs.get('if_file'))
+        if_file = self._resolve_file_path(state, if_file)
+        fc_file = T.cast(T.Optional[T.Union[str, mesonlib.File]], kwargs.get('fc_file'))
+        fc_file = self._resolve_file_path(state, fc_file)
+
+        install = T.cast(bool, kwargs['install'])
+        install_dir = T.cast(T.Optional[str], kwargs.get('install_dir'))
+        if install and install_dir is None:
+            install_dir = os.path.join(sharedir, 'packages')
+
+        name = T.cast(T.Optional[str], kwargs['name'])
+        mls = T.cast(bool, kwargs.get('mls'))
+        type = T.cast(T.Optional[str], kwargs['type'])
+        distro = T.cast(T.Optional[str], kwargs['distro'])
+        direct_initrc = T.cast(T.Optional[bool], kwargs['direct_initrc'])
+
+        pp_cmd: T.List[T.Union[str, build.BuildTarget, build.CustomTarget,
+                               build.CustomTargetIndex, ExternalProgram, mesonlib.File]] = []
+        pp_cmd.extend(state.environment.get_build_command())
+        pp_cmd.extend([
+            '--internal', 'selinux',
+            '--quiet',
+            '--output', '@OUTPUT@',
+            '--depfile', '@DEPFILE@',
+            '--private-dir', '@PRIVATE_DIR@',
+            '--te', te_file,
+        ])
+        if if_file:
+            pp_cmd.extend(['--if', if_file])
+        if fc_file:
+            pp_cmd.extend(['--fc', fc_file])
+        if name:
+            pp_cmd.extend(['--name', name])
+        if mls:
+            pp_cmd.extend(['--mls'])
+        if type:
+            pp_cmd.extend(['--type', type])
+        if distro:
+            pp_cmd.extend(['--distro', distro])
+        if direct_initrc:
+            pp_cmd.extend(['--direct-initrc'])
+
+        env = EnvironmentVariables()
+        env.set('M4', [tools['m4'].get_path()])
+        env.set('CHECKMODULE', [tools['checkmodule'].get_path()])
+        env.set('SEMOD_PKG', [tools['semodule_package'].get_path()])
```

Comment:
Uhhh.

Do *not* pass anything as an environment variable to a target. Environment variable support is a *hack* because ninja doesn't actually support them so we need to take `meson --internal selinux ...` and pickle it int SHA1.dat and run `meson --internal exe --unpickle SHA1.dat`. The real command line is hidden from ninja.